### ISSUE TITLE
Requirement file

### ISSUE THREAD

**Issue by arshiaesll** [wujx2001/QwT#3]
Can you please share the versions of the libraries working with your project, as I think the latest version are not compatible. Thank you

**Comment by heekhero** [comment]
Hi, sorry for the late reply. 

### For classification and CLIP experiments, use the following environments:

absl-py==2.1.0
accelerate==1.3.0
annotated-types==0.7.0
astunparse==1.6.3
braceexpand==0.1.7
Brotli @ file:///croot/brotli-split_1714483155106/work
calflops==0.3.2
certifi @ file:///croot/certifi_1725551672989/work/certifi
charset-normalizer @ file:///croot/charset-normalizer_1721748349566/work
clip @ git+https://github.com/openai/CLIP.git@dcba3cb2e2827b402d2701e7e1c7d9fed8a20ef1
cloudpickle==3.0.0
coloredlogs==15.0.1
contourpy==1.3.0
cycler==0.12.1
Cython @ file:///croot/cython_1727769790504/work
et-xmlfile==1.1.0
filelock @ file:///croot/filelock_1700591183607/work
flatbuffers==24.3.25
fonttools==4.54.1
fsspec==2024.9.0
ftfy==6.3.1
fvcore==0.1.5.post20221221
gast==0.6.0
gmpy2 @ file:///tmp/build/80754af9/gmpy2_1645438755360/work
google-pasta==0.2.0
graphsurgeon==0.4.5
grpcio==1.66.1
h5py==3.12.1
huggingface-hub==0.25.2
humanfriendly==10.0
idna @ file:///croot/idna_1714398848350/work
importlib_metadata==8.5.0
importlib_resources==6.4.5
iopath==0.1.10
Jinja2 @ file:///croot/jinja2_1716993405101/work
joblib==1.4.2
keras==3.6.0
kiwisolver==1.4.7
libclang==18.1.1
Mako==1.3.5
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe @ file:///croot/markupsafe_1704205993651/work
matplotlib==3.9.2
mdurl==0.1.2
mkl-service==2.4.0
mkl_fft @ file:///croot/mkl_fft_1725370245198/work
mkl_random @ file:///croot/mkl_random_1725370241878/work
ml-dtypes==0.4.1
mpmath @ file:///croot/mpmath_1690848262763/work
namex==0.0.8
networkx @ file:///croot/networkx_1717597493534/work
ninja==1.11.1.1
norm-dist-cuda==2.0.0
numpy @ file:///croot/numpy_and_numpy_base_1708638617955/work/dist/numpy-1.26.4-cp39-cp39-linux_x86_64.whl#sha256=6094eeedd869502faa0fd0a8c5ad3a70c5779be06ddd1feb7627e5c212fac420
nvidia-modelopt==0.17.0
nvidia-pyindex==1.0.9
onnx==1.17.0
onnx-graphsurgeon==0.5.2
onnx-simplifier==0.4.36
onnxruntime==1.19.2
openpyxl==3.1.5
opt_einsum==3.4.0
optree==0.13.1
packaging==24.1
pandas==2.2.3
pillow @ file:///croot/pillow_1721059439630/work
platformdirs==4.3.6
portalocker==3.1.1
prettytable==3.11.0
protobuf==3.20.3
psutil==6.1.1
pycocotools @ git+https://github.com/cocodataset/cocoapi.git@8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9#subdirectory=PythonAPI
pycuda==2024.1.2
pydantic==2.9.2
pydantic_core==2.23.4
Pygments==2.18.0
pyparsing==3.2.0
PySocks @ file:///tmp/build/80754af9/pysocks_1605305812635/work
python-dateutil==2.9.0.post0
pytools==2024.1.14
pytorch-quantization==2.1.2
pytz==2024.2
PyYAML @ file:///croot/pyyaml_1698096049011/work
regex==2024.9.11
requests @ file:///croot/requests_1721410876868/work
rich==13.8.1
safetensors==0.4.5
scikit-learn==1.5.2
scipy @ file:///croot/scipy_1710947333060/work/dist/scipy-1.12.0-cp39-cp39-linux_x86_64.whl#sha256=086bc9ddf320acb93b565ce64bf36f51436fe07a0b6b7fd6a8b38159f6ab34d5
six==1.16.0
sphinx_glpi_theme==0.6
sympy @ file:///croot/sympy_1724938189289/work
tabulate==0.9.0
tensorboard==2.18.0
tensorboard-data-server==0.7.2
tensorflow==2.18.0
tensorflow-io-gcs-filesystem==0.37.1
tensorrt @ file:///data/fumh/PETL/TensorRT/TensorRT-10.4.0.26/python/tensorrt-10.4.0-cp39-none-linux_x86_64.whl#sha256=da4d248d4c9dd0fe9cde155f611e5ba84bcab5851e8f1be2d8ac3797bd727b8d
termcolor==2.5.0
thop==0.1.1.post2209072238
threadpoolctl==3.5.0
timm==0.4.12
tokenizers==0.21.0
torch==2.4.0
torchaudio==2.4.0
torchinfo==1.8.0
torchvision==0.19.0
tqdm==4.66.5
transformers==4.48.1
triton==3.0.0
typing_extensions @ file:///croot/typing_extensions_1715268824938/work
tzdata==2024.2
urllib3 @ file:///croot/urllib3_1718912636303/work
wcwidth==0.2.13
webdataset==0.2.100
Werkzeug==3.0.4
wrapt==1.16.0
yacs==0.1.8
zipp==3.20.2


### For detection, use the following environments:

absl-py==2.1.0
addict==2.4.0
albumentations==1.4.0
aliyun-python-sdk-core==2.16.0
aliyun-python-sdk-kms==2.16.5
apex==0.1
appdirs==1.4.4
asynctest==0.13.0
attrs==23.2.0
certifi==2024.2.2
cffi==1.17.1
charset-normalizer==3.3.2
cityscapesScripts==2.2.2
click==8.1.7
codecov==2.1.13
colorama==0.4.6
coloredlogs==15.0.1
contourpy==1.1.1
coverage==7.4.3
crcmod==1.7
cryptography==43.0.3
cycler==0.12.1
Cython==0.29.33
et-xmlfile==1.1.0
exceptiongroup==1.2.0
filelock==3.14.0
flake8==7.0.0
fonttools==4.49.0
fsspec==2024.2.0
huggingface-hub==0.21.3
humanfriendly==10.0
idna==3.6
imagecorruptions==1.1.2
imageio==2.34.0
importlib-metadata==7.0.1
importlib_resources==6.1.2
iniconfig==2.0.0
interrogate==1.5.0
isort==4.3.21
jmespath==0.10.0
joblib==1.3.2
kiwisolver==1.4.5
kwarray==0.6.17
lazy_loader==0.3
Markdown==3.7
markdown-it-py==3.0.0
matplotlib==3.7.5
mccabe==0.7.0
mdurl==0.1.2
mmcv-full==1.3.0
-e /data/fumh/Quant/RepQ-ViT/detection
mmlvis==10.5.3
mmpycocotools==12.0.3
-e /data/fumh/PETL/MONA/Swin-Transformer-Semantic-Segmentation
model-index==0.1.11
networkx==3.1
ninja==1.11.1.1
norm-dist-cuda==2.0.0
numpy==1.23.5
onnx==1.7.0
onnx-simplifier==0.4.36
onnxoptimizer==0.3.13
onnxruntime==1.5.1
onnxsim==0.4.36
opencv-python==4.9.0.80
opencv-python-headless==4.9.0.80
opendatalab==0.0.10
openmim==0.3.9
openpyxl==3.1.5
openxlab==0.1.2
ordered-set==4.1.0
oss2==2.17.0
packaging==24.1
pandas==2.0.3
pillow==10.2.0
platformdirs==4.2.0
pluggy==1.4.0
prettytable==3.11.0
protobuf==3.19.0
py==1.11.0
pycodestyle==2.11.1
pycparser==2.22
pycryptodome==3.21.0
pyflakes==3.2.0
Pygments==2.18.0
pyparsing==3.1.1
pyquaternion==0.9.9
pytest==8.0.2
python-dateutil==2.8.2
pytorch-quantization==2.2.1
pytz==2023.4
PyWavelets==1.4.1
PyYAML==6.0.1
qudida==0.0.4
requests==2.28.2
rich==13.4.2
safetensors==0.4.2
scikit-image==0.21.0
scikit-learn==1.3.2
scipy==1.10.1
seaborn==0.13.2
six==1.16.0
sphinx_glpi_theme==0.6
tabulate==0.9.0
terminaltables==3.1.10
threadpoolctl==3.3.0
tifffile==2023.7.10
timm==0.9.16
toml==0.10.2
tomli==2.0.1
torch==1.8.1+cu111
torchaudio==0.8.1
torchvision==0.9.1+cu111
tqdm==4.65.2
typing==3.7.4.3
typing_extensions==4.10.0
tzdata==2024.1
ubelt==1.3.4
urllib3==1.26.20
wcwidth==0.2.13
xdoctest==1.1.3
yapf==0.40.1
zipp==3.17.0

We are aware that this may differ slightly from the instructions in the README, so please take this as a reference.
